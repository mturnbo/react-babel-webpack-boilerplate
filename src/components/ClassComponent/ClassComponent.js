import React from 'react';
import PropTypes from 'prop-types';
import Button from 'components/Button';
import './ClassComponent.scss';

class ClassComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data1: 'data1',
      data2: 'data2',
      data3: 'data3'
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    return this.state.data1 !== nextState.data1 && this.state.data2 !== nextState.data2;
  }

  render() {
    const buttonLabel = `${this.props.title} Button`;

    return (
      <div className="class-component">
        <div className="header">{this.props.title}</div>
        <div className="content">{this.props.content}</div>
        <div className="button-container">
          <Button label={buttonLabel} />
        </div>
      </div>
    );
  }
};

ClassComponent.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired
};

export default ClassComponent;
