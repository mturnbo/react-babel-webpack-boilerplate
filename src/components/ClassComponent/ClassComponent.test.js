import React from 'react';
import { mount } from 'enzyme';
import ClassComponent from './index';

const wrapper = mount(<ClassComponent title="Test" content="Blah blah blah" />);

test('ClassComponent Component Renders', () => {
  expect(wrapper).toMatchSnapshot();
});
