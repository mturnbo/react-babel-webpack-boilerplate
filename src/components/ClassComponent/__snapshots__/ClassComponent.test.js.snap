// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassComponent Component Renders 1`] = `
<ClassComponent
  content="Blah blah blah"
  title="Test"
>
  <div
    className="class-component"
  >
    <div
      className="header"
    >
      Test
    </div>
    <div
      className="content"
    >
      Blah blah blah
    </div>
    <div
      className="button-container"
    >
      <Button
        label="Test Button"
        onClick={[Function]}
      >
        <button
          className="btn"
          onClick={[Function]}
        >
          Test Button
        </button>
      </Button>
    </div>
  </div>
</ClassComponent>
`;
